js/app.js

```diff
@@ -0,0 +1,157 @@
+
+
+/**-------------- Cached Element References ---------- */
+let squares = document.querySelectorAll('.squares')
+let messageEl = document.querySelector('#message')
+let replay = document.querySelector('#replay-button')
+let board = document.querySelector('.board')
+popover = new bootstrap.Popover(document.querySelector('#info-Btn'), {
+  container: 'body'
+})
+
+/**------------------- Event Listeners --------------- */
+replay.addEventListener('click', init)
+
+board.addEventListener('click', handleClick)
+
+/**---------------------- Functions ------------------ */
+init()
+
+function init() {
+  messageEl.textContent = 'Click a Square to Start'
+  boardArray = [null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null]
+  winner = null
+  playerTurn = 1
+  replay.setAttribute('hidden', true)
+  messageEl.className = ""
+  render()
+}
+
+function render() {
+  let squareColor
+  boardArray.forEach((square, idx) => {
+    if (square === 1) {
+      squareColor = '#AC0011'
+      squares[idx].style.opacity = 1
+    } else if (square === -1) {
+      squareColor = '#0073BF'
+      squares[idx].style.opacity = 1
+    } else if (square === null) {
+      squareColor = 'white'
+      squares[idx].style.opacity = .2
+    }
+    squares[idx].style.backgroundColor = squareColor
+  })
+  // State of game:
+  if (!winner) {
+    messageEl.textContent = `${playerTurn === 1 ? " Red's" : " Blue's"} turn`
+    messageEl.style.textShadow = `${playerTurn === 1 || playerTurn === -1 ?  '0 0 10px black, 0 0 15px black, 0 0 20px black, 0px 0px 30px black, 0 0 40px' : ''}`
+    messageEl.style.color = `${playerTurn === 1 ? "#CC2727" : "#0073BF"}`
+  } else if (winner === "T") {
+    messageEl.textContent = "Tie! Let's try again!"
+    messageEl.style.color = '#667D83'
+    messageEl.style.textShadow = `${playerTurn === 1 || playerTurn === -1 ?  '10px 0 10px black, 20px 0 20px black, 30pxpx 0px 30px black' : ''}`
+  } else {
+    messageEl.textContent = `Congratulations, ${winner === 1 ? 'Red' : 'Blue'} Wins!!`
+    messageEl.style.color = `${winner === 1 ? '#CC2727' : '#0073BF'}`
+    confetti.start(1500)
+    messageEl.className = `${winner === 1 || winner === -1 ? 'animate__heartBeat' : ''}`
+    const tadaAudio = new Audio(`../Sounds/tada.mp3`)
+    tadaAudio.volume = .1
+    tadaAudio.play()
+  }
+}
+
+function handleClick(evt) {
+  if (evt.target === board) {
+    return 
+  } 
+  let id = parseInt(evt.target.id.replace('sq', ''))
+  if (boardArray[id] === -1 || boardArray[id] === 1) {
+    return
+  } else if (winner !== null) {
+    return
+  } else {
+    if (boardArray[id] === null) {
+      chipPosition(id)
+    }
+  }
+  replay.removeAttribute('hidden')
+  playerTurn *= -1
+  winner = getWinner()
+  render()
+}
+
+function chipPosition(id) {
```

Comment:
All of the `return`s out of this `chipPosition` function are unnecessary since all of the code for this function is contained inside an `if ... else if ... else` block.